packages/esm-patient-forms-app/src/htmlformentry/html-form-entry.workspace.tsx

```diff
@@ -25,25 +25,26 @@ const HtmlFormEntry: React.FC<HtmlFormEntryComponentProps> = ({
 
   // urls for entering a new form and editing an existing form; note that we specify the returnUrl as post-message:close-workspace,
   // which tells HFE-UI to send a message to the parent window to close the workspace when the form is saved or cancelled
-  const url = `${window.openmrsBase}/htmlformentryui/htmlform/${
-    htmlForm.formUiPage
-  }.page?patientId=${patientUuid}&visitId=${visitUuid ?? currentVisit?.uuid ?? null}&definitionUiResource=${
-    htmlForm.formUiResource
-  }&returnUrl=post-message:close-workspace`;
-  const urlWithEncounter = `${window.openmrsBase}/htmlformentryui/htmlform/${
-    htmlForm.formEditUiPage
-  }.page?patientId=${patientUuid}&visitId=${
-    visitUuid ?? currentVisit?.uuid ?? null
-  }&encounterId=${encounterUuid}&definitionUiResource=${
-    htmlForm.formUiResource
-  }&returnUrl=post-message:close-workspace`;
 
-  const showFormAndLoadedData = formInfo && patientUuid && patient;
+  const uiPage = encounterUuid ? htmlForm.formEditUiPage : htmlForm.formUiPage;
+  const url = `${window.openmrsBase}/htmlformentryui/htmlform/${uiPage}.page?`;
+  const searchParams = new URLSearchParams();
+  searchParams.append('patientId', patientUuid);
+  if (visitUuid || currentVisit?.uuid) {
+    searchParams.append('visitId', visitUuid ?? currentVisit?.uuid);
+  }
+  if (encounterUuid) {
+    searchParams.append('encounterId', encounterUuid);
+  }
+  searchParams.append('definitionUiResource', htmlForm.formUiResource);
+  searchParams.append('returnUrl', 'post-message:close-workspace');
+
+  const showFormAndLoadedData = formInfo && patientUuid;
```

Comment:
This is the main change that was needed.  There is no patient available anywhere at this point, nor is one needed.  I just removed the check on patient as patientUuid is sufficient.